drivers/sx127x/sx127x_getset.c

```diff
@@ -0,0 +1,735 @@
+    switch (dev->driver_type) {
+    case SX1276_TYPE:
+        return (sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG2) &
+                SX1276_RF_LORA_MODEMCONFIG2_RXPAYLOADCRC_MASK);
+    case SX1272_TYPE:
+        return (sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG1) &
+                SX1272_RF_LORA_MODEMCONFIG1_RXPAYLOADCRC_MASK);
+    }
+
+    return 0; /* should not be reached */
+}
+
+void sx127x_set_crc(sx127x_t *dev, bool crc)
+{
+    dev->settings.lora.crc_on = crc;
+    switch (dev->driver_type) {
+    case SX1276_TYPE:
+    {
+        uint8_t config2_reg = sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG2);
+        config2_reg &= SX1276_RF_LORA_MODEMCONFIG2_RXPAYLOADCRC_MASK;
+        config2_reg |= crc << 2;
+        sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG2, config2_reg);
+        break;
+    }
+    case SX1272_TYPE:
+    {
+        uint8_t config2_reg = sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG1);
+        config2_reg &= SX1272_RF_LORA_MODEMCONFIG1_RXPAYLOADCRC_MASK;
+        config2_reg |= crc << 1;
+        sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG1, config2_reg);
+        break;
+    }
+    default:
+        DEBUG("Unknown device, %d", dev->driver_type);
+        break;
+    }
+}
+
+uint8_t sx127x_get_hop_period(sx127x_t *dev)
+{
+    return sx127x_reg_read(dev, SX127X_REG_LR_HOPPERIOD);
+}
+
+void sx127x_set_hop_period(sx127x_t *dev, uint8_t hop_period)
+{
+    dev->settings.lora.freq_hop_period = hop_period;
+
+    uint8_t tmp = sx127x_reg_read(dev, SX127X_REG_LR_PLLHOP);
+    if (dev->settings.lora.freq_hop_on) {
+        tmp |= SX127X_RF_LORA_PLLHOP_FASTHOP_ON;
+        sx127x_reg_write(dev, SX127X_REG_LR_PLLHOP, tmp);
+        sx127x_reg_write(dev, SX127X_REG_LR_HOPPERIOD, hop_period);
+    }
+}
+
+bool  sx127x_get_implicit_header_mode(sx127x_t *dev)
+{
+    return dev->settings.lora.implicit_header_mode;
+}
+
+void sx127x_set_implicit_header_mode(sx127x_t *dev, bool implicit)
+{
+    dev->settings.lora.implicit_header_mode = implicit;
+
+    uint8_t config1_reg = sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG1);
+    switch (dev->driver_type) {
+    case SX1276_TYPE:
+        config1_reg &= SX1276_RF_LORA_MODEMCONFIG1_IMPLICITHEADER_MASK;
+        config1_reg |= implicit;
+        break;
+    case SX1272_TYPE:
+        config1_reg &= SX1272_RF_LORA_MODEMCONFIG1_IMPLICITHEADER_MASK;
+        config1_reg |= implicit << 2;
+        break;
+    default:
+        DEBUG("Unknown device, %d", dev->driver_type);
+        break;
+    }
+    sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG1, config1_reg);
+}
+
+uint8_t sx127x_get_payload_length(sx127x_t *dev)
+{
+    return dev->settings.lora.payload_len;
+}
+
+void sx127x_set_payload_length(sx127x_t *dev, uint8_t len)
+{
+    if (dev->settings.lora.implicit_header_mode) {
+        dev->settings.lora.payload_len = len;
+        sx127x_reg_write(dev, SX127X_REG_LR_PAYLOADLENGTH, len);
+    }
+}
+
+static inline uint8_t sx127x_get_pa_select(uint32_t channel)
+{
```

Comment:
The SX1272 transmitter mode was broken because of this change. I reverted the PA_SELECT to RFO for SX1272. For SX176 it's link to the channel used (use PABOOST in 868 MHz).
Surprisingly this is not the case [here](https://github.com/Lora-net/LoRaMac-node/blob/master/src/boards/LoRaMote/sx1272-board.c#L156), and their code shouldn't work.